pennylane/ops/op_math/adjoint_class.py

```diff
@@ -0,0 +1,279 @@
+# Copyright 2018-2022 Xanadu Quantum Technologies Inc.
+
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+
+#     http://www.apache.org/licenses/LICENSE-2.0
+
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+"""
+This submodule defines the symbolic operation that indicates the adjoint of an operator.
+"""
+
+from pennylane.operation import Operator, Operation, AdjointUndefinedError
+from pennylane.queuing import QueuingContext, QueuingError
+from pennylane.math import transpose, conj
+
+
+# pylint: disable=no-member
+class AdjointOperation(Operation):
+    """This mixin class is dynamically added to an ``Adjoint`` instance if the provided base class is an ``Operation``.
+
+    .. warning::
+        This mixin class should never be initialized independent of ``Adjoint``.
+
+    Overriding the dunder method ``__new__`` in ``Adjoint`` allows us to customize the creation of an instance and dynamically
+    add in parent classes.
+    """
+
+    @property
+    def _inverse(self):
+        return self.base._inverse
+
```

Comment:
Remember to add pylint exceptions...since this class won't be there forever this should be fine